x/concentrated-liquidity/pool_test.go

```diff
@@ -296,38 +294,204 @@ func (s *KeeperTestSuite) TestCalcOutAmtGivenIn() {
+	s.Require().NoError(err)
+	currTick := cl.PriceToTick(currPrice) // 85176
+	lowerPrice := sdk.NewDec(4545)
 	s.Require().NoError(err)
-	fmt.Printf("%v pool liq pre \n", pool.Liquidity)
-	lowerTick := int64(84222)
-	upperTick := int64(86129)
-	amount0Desired := sdk.NewInt(1)
-	amount1Desired := sdk.NewInt(5000)
-
-	s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, pool.Id, s.TestAccs[0], amount0Desired, amount1Desired, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick, upperTick)
-	pool = s.App.ConcentratedLiquidityKeeper.GetPoolbyId(ctx, pool.Id)
-	fmt.Printf("%v pool liq post 1 \n", pool.Liquidity)
-
-	// tokenIn := sdk.NewCoin("eth", sdk.NewInt(133700))
-	// tokenOutDenom := "usdc"
-	// swapFee := sdk.NewDec(0)
-	// minPrice := sdk.NewDec(4500)
-	// maxPrice := sdk.NewDec(5500)
-
-	// this is a test case for swapping within the tick
-	// amountIn, err := s.App.ConcentratedLiquidityKeeper.SwapOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, minPrice, maxPrice, pool.Id)
-	// s.Require().NoError(err)
-	// fmt.Printf("%v amountIn \n", amountIn)
-	// pool = s.App.ConcentratedLiquidityKeeper.GetPoolbyId(ctx, pool.Id)
-
-	// // calculation for this is tested in TestCalcOutAmtGivenInt
-	// s.Require().Equal(sdk.NewInt(666975610), amountIn.Amount)
-
-	// s.Require().Equal(sdk.MustNewDecFromStr("1517.818895638265328110"), pool.Liquidity)
-	// // curr sqrt price and tick remains the same
-	// s.Require().Equal(sdk.MustNewDecFromStr("70.710678000000000000"), pool.CurrentSqrtPrice)
-	// s.Require().Equal(sdk.NewInt(85176), pool.CurrentTick)
+	lowerTick := cl.PriceToTick(lowerPrice) // 84222
+	upperPrice := sdk.NewDec(5500)
+	s.Require().NoError(err)
+	upperTick := cl.PriceToTick(upperPrice) // 86129
+
+	defaultAmt0 := sdk.NewInt(1000000)
+	defaultAmt1 := sdk.NewInt(5000000000)
+
+	swapFee := sdk.ZeroDec()
+
+	tests := map[string]struct {
+		positionAmount0  sdk.Int
+		positionAmount1  sdk.Int
+		addPositions     func(ctx sdk.Context, poolId uint64)
+		tokenIn          sdk.Coin
+		tokenOutDenom    string
+		priceLimit       sdk.Dec
+		expectedTokenOut sdk.Coin
+		expectedTick     sdk.Int
+		newLowerPrice    sdk.Dec
+		newUpperPrice    sdk.Dec
+		poolLiqAmount0   sdk.Int
+		poolLiqAmount1   sdk.Int
+	}{
+		//  One price range
+		//
+		//          5000
+		//  4545 -----|----- 5500
+		"single position within one tick: usdc -> eth": {
+			addPositions: func(ctx sdk.Context, poolId uint64) {
+				// add first position
+				_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(ctx, poolId, s.TestAccs[0], defaultAmt0, defaultAmt1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
+				s.Require().NoError(err)
+			},
+			tokenIn:       sdk.NewCoin("usdc", sdk.NewInt(42000000)),
+			tokenOutDenom: "eth",
+			priceLimit:    sdk.NewDec(5004),
+			// we expect to put 42 usdc in and in return get .008398 eth back
+			// due to limited liquidity, we actually put in 41.99 usdc and in return get .008396 eth back
```

Comment:
is it actually due to limited liquidity or is it because of truncations and precision loses?

Reply:
Likely the latter, will change this